cmd/conformance-tester/pkg/clients/client_api.go

```diff
@@ -0,0 +1,401 @@
+			return nil, fmt.Errorf("failed to assign SSH key to cluster: %w", err)
+		}
+	}
+
+	log.Infof("Successfully created cluster %s", crCluster.Name)
+	return crCluster, nil
+}
+
+func (c *apiClient) CreateNodeDeployments(ctx context.Context, log *zap.SugaredLogger, scenario scenarios.Scenario, clusterName string) error {
+	nodeDeploymentGetParams := &projectclient.ListNodeDeploymentsParams{
+		Context:   ctx,
+		ProjectID: c.opts.KubermaticProject,
+		ClusterID: clusterName,
+		DC:        c.opts.Seed.Name,
+	}
+	utils.SetupParams(nil, nodeDeploymentGetParams, 5*time.Second, 1*time.Minute)
+
+	log.Info("Getting existing NodeDeployments…")
+	resp, err := c.opts.KubermaticClient.Project.ListNodeDeployments(nodeDeploymentGetParams, c.opts.KubermaticAuthenticator)
+	if err != nil {
+		return fmt.Errorf("failed to get existing NodeDeployments: %w", err)
+	}
+
+	existingReplicas := 0
+	for _, nodeDeployment := range resp.Payload {
+		existingReplicas += int(*nodeDeployment.Spec.Replicas)
+	}
+	log.Infof("Found %d pre-existing node replicas", existingReplicas)
+
+	nodeCount := c.opts.NodeCount - existingReplicas
+	if nodeCount < 0 {
+		return fmt.Errorf("found %d existing replicas and want %d, scaledown not supported", existingReplicas, c.opts.NodeCount)
+	}
+	if nodeCount == 0 {
+		return nil
+	}
+
+	log.Info("Preparing NodeDeployments")
+	var nodeDeployments []apimodels.NodeDeployment
+	if err := wait.PollImmediate(10*time.Second, time.Minute, func() (bool, error) {
+		var err error
+		nodeDeployments, err = scenario.NodeDeployments(ctx, nodeCount, c.opts.Secrets)
+		if err != nil {
+			log.Warnw("Getting NodeDeployments from scenario failed", zap.Error(err))
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("didn't get NodeDeployments from scenario within a minute: %w", err)
+	}
+
+	log.Info("Creating NodeDeployments…")
+	for _, nd := range nodeDeployments {
+		params := &projectclient.CreateNodeDeploymentParams{
+			Context:   ctx,
+			ProjectID: c.opts.KubermaticProject,
+			ClusterID: clusterName,
+			DC:        c.opts.Seed.Name,
+			Body:      &nd,
+		}
+		utils.SetupParams(nil, params, 5*time.Second, 1*time.Minute, http.StatusConflict)
+
+		if _, err := c.opts.KubermaticClient.Project.CreateNodeDeployment(params, c.opts.KubermaticAuthenticator); err != nil {
+			return fmt.Errorf("failed to create NodeDeployment %s: %w", nd.Name, err)
+		}
+	}
+
+	log.Infof("Successfully created %d NodeDeployments", nodeCount)
+	return nil
+}
+
+func (c *apiClient) DeleteCluster(ctx context.Context, log *zap.SugaredLogger, report *reporters.JUnitTestSuite, cluster *kubermaticv1.Cluster) error {
+	deleteTimeout := 15 * time.Minute
+	if cluster.Spec.Cloud.Azure != nil {
+		// 15 Minutes are not enough for Azure
+		deleteTimeout = 30 * time.Minute
+	}
+
+	if err := util.JUnitWrapper(
+		"[Kubermatic] Delete cluster",
+		report,
+		func() error {
```

Comment:
is there any reason we're doing the junit wrapper within the `DeleteCluster` implementation? Wouldn't outside and wrapping around the various implementations be better?

Reply:
Moved it.